Hmac.js

```diff
@@ -48,12 +54,42 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} [options.useCache=true] - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    const cacheKey = `sign-${base64url.encode(data)}`;
+    if(useCache) {
+      const signature = this.cache.get(cacheKey);
+      if(signature) {
+        return signature;
+      }
+    }
+
     const {id: keyId, kmsClient, capability, invocationSigner} = this;
-    return kmsClient.sign({keyId, data, capability, invocationSigner});
+    const promise = kmsClient.sign({keyId, data, capability, invocationSigner});
+
+    if(useCache) {
+      // 1. Set promise in cache
+      this.cache.set(cacheKey, promise);
+
+      // 2. Schedule cache pruning if not already scheduled
+      if(!this._pruneCacheTimer) {
+        this._pruneCacheTimer = setTimeout(
+          () => this._pruneCache(), MAX_CACHE_AGE);
```

Comment:
Does this work?

```suggestion
        this._pruneCacheTimer = setTimeout(this._pruneCache, MAX_CACHE_AGE);
```

Reply:
Nope, my suggestion doesn't work doing `.bind()` https://repl.it/repls/FamiliarInexperiencedTrapezoid